sdk/Lusid.Sdk/Client/ApiClient.cs

```diff
@@ -136,15 +136,11 @@ internal object Deserialize(IRestResponse response, Type type)
                 return Convert.ChangeType(response.Content, type);
             }
 
-            // at this point, it must be a model (json)
-            try
-            {
-                return JsonConvert.DeserializeObject(response.Content, type, _serializerSettings);
-            }
-            catch (Exception e)
-            {
-                throw new ApiException(500, e.Message);
-            }
+            // If for some reason the response is a null, empty, or whitespace, the serialisation should fail.
+            if (string.IsNullOrWhiteSpace(response.Content))
+                throw new DeserializationException(response, new Exception($"API Response Content was invalid: '{response.Content}'"));
+            
+            return JsonConvert.DeserializeObject(response.Content, type, _serializerSettings);
```

Comment:
I'm not convinced that we want to make this change. If we can handle this scenario with a custom exception handler why modify the generated template? Might be better to post an issue on the OpenApi Generator repo instead.

These changes make upgrading the version of OpenApi much more difficult as we need to migrate them over to a potentially modified template and it isn't always clear what their significance is.

Reply:
We can't really handle empty or whitespace responses in the exception handler robustly. If we pass an empty string through the Deserializer, we end up with no error thrown and just a 'null' return value, which then becomes a null API method response. So not sure what to do about that in the nearest future.

I do agree raising an issue with Open API